src/core/common/message.cpp

```diff
@@ -404,7 +404,7 @@ otError Message::Append(const void *aBuf, uint16_t aLength)
 
 otError Message::AppendTlv(const Tlv &aTlv)
 {
-    return Append(&aTlv, aTlv.GetSize());
+    return Append(&aTlv, static_cast<uint16_t>(aTlv.GetSize()));
```

Comment:
If GetSize() really overflow uint16_t, it seems this will still cause issues. Shall we assert it is not greater than 65535?

Reply:
The primary issue this PR addresses is processing received TLVs that have length > 65535. I believe the vast majority of issues should be caught in `Tlv::Find()` before we get to `Message::AppendTlv()`. In any case, I have added the assert here to be safe.